### PR TITLE
Song play controls

### PR DESCRIPTION
**Pull request by gracewainaina**

### CODE REVIEW

**Comment by gracewainaina** [comment]
Demo for play next song or previous song can be found here: https://github.com/gracewainaina/Musicca/blob/song-play-controls/Screenshots/play_next_previous.mp4

Demo for pausing/ playing each song can be found here: https://github.com/gracewainaina/Musicca/blob/song-play-controls/Screenshots/play_pause_controls.mp4

File: Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java [hunk 1]
```diff
@@ -78,20 +84,61 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    songPosition--;
+                    setSongView();
+                }
+            }
+        });
+
+        ivNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast.makeText(SongPlaylistActivity.this, "Clicked!", Toast.LENGTH_SHORT);
```
Comment (xuyueee) [inline Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java]: Why do you need this toast?

File: Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java [hunk 2]
```diff
@@ -48,14 +52,16 @@
...
-    private String albumUrl;
     private String playlistObjectId;
-    private String songObjectId;
 
     private static final String CLIENT_ID = "22793b7728c54470b8d117506f9574c5";
     private static final String REDIRECT_URI = "com.musicca://callback";
     private SpotifyAppRemote mSpotifyAppRemote;
 
+    // boolean to determine is song is being played for the first time, on resume or on pause
+    private boolean isPlay = false;
```
Comment (xuyueee) [inline Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java]: The name is a little confusing. What does this do exactly? You might want to rename it to make it more clear.

**Review by xuyueee — COMMENTED** [review]
Looks really good!

File: Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java [hunk 3]
```diff
@@ -78,20 +84,134 @@ protected void onCreate(Bundle savedInstanceState) {
         ivPlayPause = findViewById(R.id.ivPlayPause);
         ivNext = findViewById(R.id.ivNext);
 
-        songObjectId = getIntent().getStringExtra(EXTRA_SONGOBJECTID);
+<<<<<<< HEAD
+<<<<<<< HEAD
+<<<<<<< HEAD
+=======
+>>>>>>> d632048cbc776aadb255fa324e3b537c1c8edd96
```
Comment (xuyueee) [inline Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java]: There seem to be a lot of merge conflicts that are not resolved in this current version.

File: Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java (hunk 2 above)
Comment (gracewainaina) [reply Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java]: I have renamed the booleans to isSongPlaying and isSongPaused

File: Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java (hunk 3 above)
Comment (gracewainaina) [reply Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java]: I had not realised there were conflicts, I have managed fixed them though

File: Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java (hunk 1 above)
Comment (gracewainaina) [reply Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java]: This toast is not necessary anymore. I was using it to debug, but should have removed them when I finished

**Pull request merged**